File: baekjoon_all/11000+/boj_11758.cpp
// Solve 2022-07-13
// Update 2024-02-18

#include <bits/stdc++.h>
using namespace std;

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL); // boj_15552.cpp
#define SETPRECISION(n) cout << fixed;cout.precision(n); // boj_1008.cpp
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
using ll = long long;

struct Point{
    ll x, y;

    Point() {}

    Point(ll x, ll y) : x(x), y(y) {}

    Point operator-(const Point &p) const {
        return { x - p.x, y - p.y };
    }

    ll get_cross(const Point &p) {
        return x * p.y - p.x * y;
    }

    ll get_ccw(const Point &p1, const Point &p2) {
        Point v1 = p1 - *this;
        Point v2 = p2 - *this;
        return v1.get_cross(v2);
    }

    int get_ccw_sign(const Point &p1, const Point &p2) {
        ll ccw = get_ccw(p1, p2);

        if (ccw > 0) return 1;
        if (ccw < 0) return -1;
        return 0;
    }
};

int main() {
    FASTIO;

    Point points[3];

    for (int i = 0; i < 3; i++) {
        cin >> points[i].x >> points[i].y;
    }

    cout << points[0].get_ccw_sign(points[1], points[2]) << '\n';

    return 0;
}

File: baekjoon_all/11000+/boj_11758.java
// Solve 2024-02-18

// 백준에 제출할 때는 class 이름을 Main으로 설정해야 한다.

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class boj_11758 {

    static class Point {
        int x, y;

        Point(int x, int y) {
            this.x = x;
            this.y = y;
        }

        Point subtract(Point p) {
            return new Point(this.x - p.x, this.y - p.y);
        }

        int getCross(Point p) {
            return this.x * p.y - p.x * this.y;
        }

        int getCcw(Point p1, Point p2) {
            return p1.subtract(this).getCross(p2.subtract(this));
        }

        int getCcwSign(Point p1, Point p2) {
            int ccw = getCcw(p1, p2);

            if (ccw > 0) return 1;
            if (ccw < 0) return -1;
            return 0;
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        Point[] points = new Point[3];

        for (int i = 0; i < 3; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            points[i] = new Point(x, y);
        }

        System.out.println(points[0].getCcwSign(points[1], points[2]));

        br.close();
    }

}
